src/main/groovy/edu/oregonstate/mist/api/Resource.groovy

```diff
@@ -1,14 +1,35 @@
 package edu.oregonstate.mist.api
 
+import javax.ws.rs.core.Context
 import javax.ws.rs.core.Response
 import javax.ws.rs.core.Response.ResponseBuilder
+import org.apache.http.client.utils.URIBuilder
+import javax.ws.rs.core.UriInfo
 
 /**
  * Abstract class for reusing common response messages.
  */
 abstract class Resource {
     protected static Properties properties = new Properties()
 
+    /**
+     * Default page number used in pagination
+     */
+    public static final Integer DEFAULT_PAGE_NUMBER = 1
+
+    /**
+     * Default page size used in pagination
+     */
+    public static final Integer DEFAULT_PAGE_SIZE = 10
+
+    @Context
+    UriInfo uriInfo
+
+    /**
+     * URI used to provide jsonapi pagination links.
+     */
+    private URI endpointUri
+
     public static loadProperties() {
```

Comment:
Good catch. Would you mind creating a ticket for this? The code in this package is part of the skeleton, so we should handle it in that repo.